tasks/dispense.js

```diff
@@ -0,0 +1,59 @@
+const { task } = require('hardhat/config');
+const { getBalances, getSigners, getContract } = require('../util');
+const rifFaucetAbi = require('../abi/rifFaucet.json');
+
+// connects to test RIF faucet and dispenses the tokens to
+// supplied wallets
+async function getTestRifs(wallets) {
+  const rifFaucet = await getContract({
+    name: 'RIFFaucet',
+    abi: rifFaucetAbi,
+  });
+
+  const txs = await Promise.allSettled(
+    wallets.map(async (wallet) => {
+      const txRequest = await rifFaucet
+        .connect(wallet)
+        .dispense(wallet.address);
+      return txRequest.wait();
+    }),
+  );
+  // fulfilled transactions
+  txs
+    .filter((tx) => tx.status === 'fulfilled')
+    .forEach((tx) => {
+      console.log(`Topped up RIF token balance for ${tx.value.from}`);
+    });
+  // rejected transactions
+  txs
+    .filter((tx) => tx.status === 'rejected')
+    .forEach((tx) => {
+      console.log(
+        `Couldn't dispense RIF tokens for ${tx.reason.transaction.from}`,
+      );
+    });
+}
```

Comment:
:+1:

Reply:
perhaps there should also be another for "dispensing" testnet RNS as well, for the NFT tests.